clearwater-socket-factory.md

```diff
@@ -1,34 +1,44 @@
 
 To request a new TCP connection from the factory a client should do the following:
 
-* Create a new UNIX socket and connect to `/tmp/clearwater_mgmt_namespace_socket`.
+* Create a new UNIX socket and connect to `/tmp/clearwater_mgmt_namespace_socket` or `/tmp/clearwater_signaling_namespace_socket`.
 * Create a target string of the form `<host>:<port>` and send this in the UNIX socket using `send`.  `<host>` may be an IPv4 address, or a hostname that resolves to one or more IPv4 addresses.
 * The daemon will now establish a TCP connection to the specified target and sends a message over the UNIX socket containing the TCP socket in the control data.
 * The client receives this message and iterates though the control headers on the message until it finds one containing a socket (where the level is `SOL_SOCKET` and the type is `SCM_RIGHTS`).
 * The client can treat the received socket as it would any normal connected TCP socket.
 
-See the [example client](clearwater-socket-factory/test_client.c) for more details.
+See the [example client](clearwater-socket-factory/test_client.cpp) for more details.
 
 ### Error Conditions
 
-If the daemon is asked to connect to a host other than the configured "allowed host" it assumes that the allowed host must have changed. It exits immediately so that it can be restarted and pick up the new value.
+If the daemon is asked to connect to a host other than the configured "allowed hosts" it assumes that the allowed host must have changed. It exits immediately so that it can be restarted and pick up the new value.
 
 If the daemon encounters any other errors it will indicate this to the client by either:
 
 * Closing the connection to client.
 * Sending the client a message containing an invalid (negative) file descriptor.
 
+### Configuration
+
+`clearwater-socket-factory` has 3 configuration options:
+
+* signaling-namespace - this is the name of the signaling namespace.
+* management-allowed-hosts - this is a comma separated lists of hosts in the management network that clearwater-socket-factory is allowed to provide sockets for.
+* signaling-allowed-hosts - this is a comma separated lists of hosts in the signaling network that clearwater-socket-factory is allowed to provide sockets for.
+
+`clearwater-socket-factory` has an upstart script which is built into its Debian package. When deployed with this upstart script, the process will continually respawn when it exits. The upstart script reads signaling_namespace in /etc/clearwater/config to use as the signaling-namespace option, and it builds up lists of allowed hosts by reading each line of each file in /etc/clearwater-socket-factory/signaling.d and /etc/clearwater-socket-factory/management.d to use as the signaling-allowed-hosts and management-allowed-hosts options. If these directories don't exist, or are empty, the whitelist is empty.
```

Comment:
If this were me, I'd use backticks a lot more to make it clear which bits are literals and which are just words.